core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -186,20 +187,32 @@ abstract class QueryTest extends SparkFunSuite {
       try {
         if (lhs.length != rhs.length) {
           false
-        } else if (!isOrdered) {
-          comp(
-            lhs.sortWith((_1, _2) => sortComp(_1, _2)),
-            rhs.sortWith((_1, _2) => sortComp(_1, _2))
-          )
         } else {
-          implicit object NullableListOrdering extends Ordering[List[Any]] {
-            override def compare(p1: List[Any], p2: List[Any]): Int =
-              p1.contains(null).compareTo(p2.contains(null))
-          }
-          comp(
-            lhs.sortBy[List[Any]](x => x),
-            rhs.sortBy[List[Any]](x => x)
+          // the result order may be different in TiDB and TiSpark, so compare result as follows
+          // 1. compare directly
```

Comment:
if we know that TiDB/TiSpark result should be ordered, why should we compare directly?

Reply:
Because now there may be different data type for the same data, for example 1/0 and True/False, there is no easy way to sort the result so that the same data stay in the same position in `lhs/rhs`. And for most of the simple test(like `MultiDataTypeTests`), although there is no `order by` clause in the query, the result order is actually the data insertion order, we can compare it directly.